pkg/network/ebpf/c/http.h

```diff
@@ -7,6 +7,26 @@
 
 #include <uapi/linux/ptrace.h>
 
+// HTTP2.0 client->server connection preface 'PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n'
+// https://httpwg.org/specs/rfc7540.html#ConnectionHeader
+#define HTTP20_PRI_LEN 24
+#if HTTP_BUFFER_SIZE < HTTP20_PRI_LEN
+#error "HTTP_BUFFER_SIZE can't less than HTTP20_PRI_LEN"
+#endif
+static __always_inline int is_http20(char *p) {
+    if ((p[0] == 'P') && (p[1] == 'R') && (p[2] == 'I') && (p[3] == ' ') && (p[4] == '*') && (p[5] == ' ')
```

Comment:
This seems like a lot of comparison, just to bail out. Could we match on something shorter, like just `PRI * `? Or do we even need this at all, since `PRI` will not match any of our request methods.

Reply:
is_http20() is called only if the buffer len == HTTP20_PRI_LEN so the impact on performance is minimal
PRI is define in the RFC, I don't think we should do a shorter pattern matching here (as it's user buffer) even if it's unlikely to have false positive here with `PRI *` only

Reply:
The number of instructions is a limitation in eBPF, so we should make sure we are getting value from new code.

Reply:
I agree with @brycekahle. Unfortunately I'm not sure if we are actually benefiting much from doing this. I think the main perfomance gain here would be one eBPF map lookup per TCP segment (that is done inside `http_process`), but still, is this gain _measurable_ in any way?

Aside from that, the optimization here may be a drop in the ocean as we really don't know how HTTP2 traffic compares to the total TLS traffic.
Hypothetically, let's say we deploy this on a kafka broker. This bail out wouldn't apply in 99% of and we would still call `http_process` for nearly all network packets, containing kafka protocol messages 🤷‍♂️ so the perfomance gains here are really hard to weigh in IMO

Reply:
In all likelihood, we are going to be parsing HTTP/2 in the near future. Then there may not even be a performance gain because we will be doing some other operations.